### PR TITLE
mgr/dashboard: converted pools list table tab to resource pages

### PR DESCRIPTION
**Pull request by syedali237**
Fixes: https://tracker.ceph.com/issues/76713

https://github.com/user-attachments/assets/2e6cbf39-a31c-4f2e-8752-9bd8f73df8c2

Screenshot From 2026-07-20 18-00-19

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)

You must only issue one Jenkins command per-comment. Jenkins does not understand
comments with more than one command.
</details>

### CODE REVIEW

**Comment by syedali237** [comment]
ok to test

**Comment by cloudbehl** [comment]
When the Quota is set, the available and the total is different values. Like in your example the quota is 4Gib and the available is 28 GiB. Which from customer stand point is confusing.  Maybe when the quota is set, we can  show total and available based on that.

image

**Comment by syedali237** [comment]
Thanks! I will discuss this with my mentor, I agree this could be more clearer..

**Comment by dnyanee1997** [comment]
Let us know onces rebased. @syedali237

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    readThroughput: '-',
+    readOps: '-',
+    readBytesRates: [] as number[],
+    writeThroughput: '-',
+    writeOps: '-',
+    writeBytesRates: [] as number[]
+  };
+
+  omittedPoolAttributes = ['cdExecuting', 'cdIsBinary', 'stats'];
+  private readonly binaryUnits = ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
```
Comment (syedali237) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts]: check for formatter service

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    minSize: '-',
+    erasureK: '-',
+    erasureM: '-',
+    erasureTotal: '-',
+    erasurePlugin: '-',
+    failureDomain: '-',
+    readThroughput: '-',
+    readOps: '-',
+    readBytesRates: [] as number[],
+    writeThroughput: '-',
```
Comment (syedali237) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts]: remove this , this can be compensated by overview card for empty values

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html [hunk 3]
```diff
@@ -1,79 +1,36 @@
...
+          <cds-dropdown
+            [label]="'Actions'"
+            [hideLabel]="true"
+            [placeholder]="'Actions'"
+            size="lg"
+            class="pool-actions-dropdown"
+            (selected)="onPoolActionSelected($event)"
+          >
+            <cds-dropdown-list [items]="poolActionItems"></cds-dropdown-list>
+          </cds-dropdown>
```
Comment (syedali237) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html]: use the menu button

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html [hunk 4]
```diff
@@ -1,79 +1,36 @@
...
-        </cd-table>
-      </ng-template>
-    </ng-container>
-  </nav>
+      <div class="pool-header-tags">
+        @if (poolDataProtection) {
+          <cds-tag class="tag-dark me-1">{{ poolDataProtection }}</cds-tag>
+        }
+        @for (app of poolApplications; track app) {
+          <cds-tag class="tag-dark me-1">{{ app }}</cds-tag>
```
Comment (syedali237) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html]: bootstrap class

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool.module.ts [hunk 5]
```diff
@@ -113,6 +120,39 @@ export class PoolModule {
...
+        path: 'overview',
+        component: PoolDetailsSectionComponent,
+        data: { breadcrumbs: 'Overview', section: 'overview' }
+      },
+      {
+        path: 'advance-properties',
+        component: PoolDetailsSectionComponent,
+        data: { breadcrumbs: 'Advance Properties', section: 'advance-properties' }
+      },
+      {
```
Comment (syedali237) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool.module.ts]: remove advance property section

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts (hunk 2 above)
Comment (syedali237) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts]: this is used for i/o summary cards

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.spec.ts [hunk 6]
```diff
@@ -1,172 +1,82 @@
...
-        stats: { anyStat: 3, otherStat: [1, 2, 3] }
-      });
-      const expectedPool = { prop1: 1, prop2: 2, prop3: 3 };
-      expect(poolDetailsComponent.poolDetails).toEqual(expectedPool);
+    it('should build sidebar items', () => {
+      expect(poolDetailsComponent.sidebarItems.map((item) => item.label)).toEqual([
+        'Details',
+        'Performance Details',
+        'Configuration',
+        'Cache Tiers Details'
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.spec.ts]: **Bug: These expected labels do not match the actual code.** The `buildSidebarItems` method in `pool-details.component.ts` builds:
```
['Overview', 'Configuration', 'Performance', 'Advance Properties']
```
but this test expects:
```
['Details', 'Performance Details', 'Configuration', 'Cache Tiers Details']
```
This test will fail. Please update to match the actual sidebar labels.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts [hunk 7]
```diff
@@ -1,81 +1,287 @@
...
+      //   label: $localize`Cache Tiers Details`,
+      //   route: [this.basePath, this.poolName, 'cache-tiers-details'],
+      //   routerLinkActiveOptions: { exact: true }
+      // }
       {
-        prop: 'cache_min_flush_age',
-        name: $localize`Min Flush Age`,
-        flexGrow: 2
-      },
+        label: $localize`Advance Properties`,
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts]: **Typo:** "Advance Properties" should be "Advanced Properties" (with a 'd'). This typo appears in the label, the route path (`advance-properties`), and the corresponding route definitions in `pool.module.ts` and the `@case` in the section component HTML. Please fix all occurrences.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts [hunk 8]
```diff
@@ -1,81 +1,287 @@
...
+    if (permissions.grafana?.read) {
+      items.push({
+        label: $localize`Performance`,
+        route: [this.basePath, this.poolName, 'performance'],
+        routerLinkActiveOptions: { exact: true }
+      });
+    }
+
+    items.push(
+      // {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts]: **Remove commented-out code.** There are two blocks of dead commented-out code here (lines 97-106 and lines 114-120). Same issue in `pool-details-section.component.ts` (lines 215-221) and `pool-details-section.component.html` (lines 74-80). Please remove all commented-out blocks before merging -- they add noise and can be found in git history if ever needed.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-cards/pool-details-capacity-protection-cards.component.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'cd-pool-details-capacity-protection-cards',
+  templateUrl: './pool-details-capacity-protection-cards.component.html',
+  styleUrls: ['./pool-details-capacity-protection-cards.component.scss'],
+  standalone: false
+})
+export class PoolDetailsCapacityProtectionCardsComponent {
+  @Input() overviewModel: any;
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-cards/pool-details-capacity-protection-cards.component.ts]: **Missing TypeScript type.** Both card components use `@Input() overviewModel: any`. Please define a `PoolOverviewModel` interface (e.g. in `pool.ts` or a new `pool-overview.model.ts`) and use it here and in the IO cards component. The `overviewModel` object built in `PoolDetailsSectionComponent.buildOverviewModel()` already has a clear shape -- extracting it as an interface will provide type safety and better IDE support.

```ts
export interface PoolOverviewModel {
  name: string;
  type: string;
  typeLabel: string;
  // ... etc
}
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-cards/pool-details-io-cards.component.scss [hunk 10]
```diff
@@ -0,0 +1,51 @@
+@use '@carbon/layout';
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-cards/pool-details-io-cards.component.scss]: **DRY: Duplicated SCSS.** This file shares ~7 identical class definitions with `pool-details-capacity-protection-cards.component.scss`:
- `.pool-overview-card`
- `.pool-overview-panels`
- `.pool-detail-panel`
- `.pool-detail-panel__subtitle`
- `.pool-detail-row`
- `@media` queries

Please extract the shared styles into a common SCSS partial (e.g. `_pool-detail-cards.scss`) and `@use` it in both component stylesheets.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-cards/pool-details-capacity-protection-cards.component.scss [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+  justify-content: space-between;
+  gap: layout.$spacing-03;
+}
+
+.pool-detail-divider {
+  border: 0;
+  border-top: 1px solid var(--cds-border-subtle-01);
+}
+
+:host ::ng-deep .pool-detail-panel cd-usage-bar {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-cards/pool-details-capacity-protection-cards.component.scss]: **`::ng-deep` is deprecated.** Use `ViewEncapsulation.None` on the component instead, as the established pattern does (see `host-resource-sidebar.component.ts`). Same applies to the `::ng-deep` in `pool-details-io-cards.component.scss` line 36.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html [hunk 12]
```diff
@@ -1,79 +1,36 @@
...
-          [autoSave]="false"
-          columnMode="flex"
-        >
-        </cd-table>
-      </ng-template>
-    </ng-container>
-  </nav>
+      <div class="pool-header-tags">
+        @if (poolDataProtection) {
+          <cds-tag class="tag-dark me-1">{{ poolDataProtection }}</cds-tag>
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html]: **Bootstrap class.** `me-1` is a Bootstrap utility class. Use the Carbon spacing class `cds-mr-2` instead, which is what the host-resource-sidebar uses for tag spacing. Same on line 16.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html [hunk 13]
```diff
@@ -1,79 +1,36 @@
...
+        @for (app of poolApplications; track app) {
+          <cds-tag class="tag-dark me-1">{{ app }}</cds-tag>
+        }
+      </div>
 
-  <div [ngbNavOutlet]="nav"></div>
-</ng-container>
+      @if (getVisiblePoolActions().length) {
+        <div class="pool-header-actions">
+          <cds-dropdown
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.html]: **Use `cds-menu-button` instead of `cds-dropdown` for actions.** A dropdown is semantically a form select input, not an action menu. The host-resource-sidebar uses `cds-menu-button` for this exact purpose -- it renders as a button with a menu of actions, which is the correct Carbon pattern.

See `host-resource-sidebar.component.html` lines 31-48 for the established pattern:
```html
<cds-menu-button
  kind="tertiary"
  size="lg"
  menuId="pool-actions-menu"
  menuAlignment="bottom-end"
  label="Actions"
  i18n-label
>
  @for (action of getVisiblePoolActions(); track action.name) {
    <cds-menu-item ...></cds-menu-item>
  }
</cds-menu-button>
```
This will also eliminate the large block of custom CSS in `pool-details.component.scss` (lines 24-79) that manually re-styles `cds-dropdown` to look like a button.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+  private buildOverviewModel(pool?: Pool, erasureProfile?: ErasureCodeProfile) {
+    const poolData = (pool as Pool) || ({} as Pool);
+    const stats = (poolData.stats || {}) as PoolStats;
+    const usageUsed = this.getStatLatest(stats.bytes_used);
+    const usageAvailable = this.getStatLatest(stats.avail_raw);
+    const usageTotal =
+      usageUsed !== null && usageAvailable !== null ? usageUsed + usageAvailable : null;
+    const readBytesRates = this.getRateSeries(stats.rd_bytes);
+    const writeBytesRates = this.getRateSeries(stats.wr_bytes);
+    const appLabels: Record<string, string> = {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts]: **DRY: `appLabels` is duplicated 3 times in this PR.** It appears here (line 146), again in `buildOverviewSections` (line 222), and in `pool-details.component.ts` (line 155). The same map also exists in `pool-list.component.ts` (line 277). Please extract this to a shared constant, e.g. in `pool.ts`:
```ts
export const POOL_APP_LABELS: Record<string, string> = {
  cephfs: $localize`File system`,
  rbd: $localize`Block`,
  rgw: $localize`Object`
};
```

Similarly, `getDataProtection()` is duplicated between `pool-details.component.ts` and this file -- extract it as a shared utility.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts [hunk 15]
```diff
@@ -0,0 +1,325 @@
...
+  private loadPool(): void {
+    if (!this.poolName) {
+      this.poolOverviewSections = [];
+      return;
+    }
+
+    forkJoin([
+      this.poolService.get(this.poolName),
+      this.poolService.getConfiguration(this.poolName),
+      this.poolService.getList(),
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.ts]: **Performance: unnecessary full list fetch.** `this.poolService.getList()` fetches the entire pool list just to find cache tier pools (line 119) and merge stats (line 121). Since you already have `this.poolService.get(this.poolName)` returning the individual pool (which includes `tiers` IDs), consider fetching only the needed tier pools individually, or restructuring to avoid the full list call. On clusters with hundreds of pools, this is an unnecessary load.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.html [hunk 16]
```diff
@@ -0,0 +1,88 @@
+@if (poolName) {
+  @switch (section) {
+    @case ('overview') {
+      <cd-resource-overview-card
+        title="Pools details"
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details-section.component.html]: **Missing i18n.** The `title="Pools details"` attribute is not marked for i18n. Use `i18n-title` and correct to `title="Pool Details"` (singular, since this is one pool's detail view).

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.scss [hunk 17]
```diff
@@ -0,0 +1,94 @@
+.pool-header-extra {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  gap: 1rem;
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.scss]: **Use Carbon layout tokens instead of raw values.** Lines like `gap: 1rem`, `gap: 0.25rem`, `min-height: 2.5rem` etc. should use Carbon spacing tokens (`layout.$spacing-05`, `layout.$spacing-02`, etc.) for consistency. Add `@use '@carbon/layout';` at the top and replace raw values throughout this file.

**Review by afreen23 — CHANGES_REQUESTED** [review]
Thanks for the work on migrating pools to the resource page pattern, @syedali237. The overall structure follows the established host-resource-page pattern well. However, there are several issues that need to be addressed before this can be merged.

**Blockers:**
1. The unit test expects sidebar labels that do not match the actual code -- the test will fail.
2. Multiple blocks of commented-out dead code must be removed.
3. Typo: "Advance Properties" should be "Advanced Properties" (route path and label).

**Key improvements needed:**
- Define a TypeScript interface for `overviewModel` instead of using `any`.
- Extract duplicated SCSS shared by both card components.
- Use `cds-menu-button` for the actions menu (matching the host sidebar pattern), not `cds-dropdown`.
- Replace `::ng-deep` with `ViewEncapsulation.None`.
- Replace bootstrap `me-1` with Carbon spacing classes (`cds-mr-2`).
- Extract duplicated `appLabels` and `getDataProtection` into shared helpers.
- Add unit tests for `PoolDetailsSectionComponent` -- it has 325 lines of logic with zero test coverage.

See inline comments for details.

**Comment by dnyanee1997** [comment]
@syedali237 , fix unit test cases on PoolList and PoolDetail component.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts (hunk 7 above)
Comment (syedali237) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts]: it was suggested to remove this section and the key value dump ?  should we go with that?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts [hunk 18]
```diff
@@ -0,0 +1,319 @@
...
+        type: 'tags'
+      },
+      {
+        label: $localize`PG Status`,
+        value: this.transformPgStatus(poolData.pg_status)
+      }
+    ];
+  }
+
+  private transformPgStatus(pgStatus: any): string {
```
Comment (syedali237) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts]: this exists in `pool-list.component.ts` , can be a shared function in both places

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts [hunk 19]
```diff
@@ -0,0 +1,319 @@
...
+      readThroughput: this.formatRate(this.getStatRate(stats.rd_bytes), true) || '',
+      readOps: this.formatRate(this.getStatRate(stats.rd), false) || '',
+      readOpsChartData,
+      writeThroughput: this.formatRate(this.getStatRate(stats.wr_bytes), true) || '',
+      writeOps: this.formatRate(this.getStatRate(stats.wr), false) || '',
+      writeOpsChartData
+    };
+  }
+
+  private getErasureProfile(
```
Comment (syedali237) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts:215]: this can also be moved into a shared service , already exists in pool-list

**Comment by cloudbehl** [comment]
Thanks for doing great work.


Can we make throughput as a graphs and add the instant values in the legends. 

We don't need to write Read OPS and Read Throughput. IF the heading is Read we can write it as Throughput and Ops. 

image

**Comment by cloudbehl** [comment]
When the type is Erasure coding, how are we showing data protection?

**Comment by syedali237** [comment]
@cloudbehl  for erasure coded

Screenshot From 2026-07-20 20-49-39


and yes I'll check and add throughput data as a graph

**Comment by syedali237** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts (hunk 7 above)
Comment (afreen23) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts]: Then why its not removed ??

Comment (syedali237) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.ts]: will remove it

**Comment by syedali237** [comment]
https://tracker.ceph.com/issues/78480 taking the throughput graph as a future enhancement

**Review by dnyanee1997 — COMMENTED** [review]
Edit Pool looks misaligned from left margin but didnt see any chnges in edit-pool list . Please re check its unrelated to this PR. 

Image

**Review by dnyanee1997 — COMMENTED** [review]
Both PoolCapacityProtectionCardComponent and PoolIoCardComponent use ViewEncapsulation.None, which leaks their CSS globally.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts (hunk 18 above)
Comment (dnyanee1997) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts]: Yes transformPgStatus is duplicated

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts [hunk 20]
```diff
@@ -212,7 +167,7 @@ export class PoolListComponent extends ListWithDetails implements OnInit {
             return this.poolService.getList();
           })
         ),
-      undefined,
+      (resp: any) => resp,
```
Comment (dnyanee1997) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts]: type ?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts [hunk 21]
```diff
@@ -0,0 +1,319 @@
...
+  }
+
+  private getRateChartData(stat: PoolStat | undefined, groupLabel: string): ChartPoint[] {
+    if (!Array.isArray(stat?.rates)) {
+      return [];
+    }
+
+    const fallbackStart = Date.now() - Math.max(stat.rates.length - 1, 0) * 60000;
+
+    return stat.rates.reduce((points: ChartPoint[], point: any, index: number) => {
```
Comment (dnyanee1997) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts]: Type ? 
Try not using any. Check other place as well

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool.module.ts [hunk 22]
```diff
@@ -113,6 +124,34 @@ export class PoolModule {
...
+    component: PoolResourceSidebarComponent,
+    data: { breadcrumbs: PoolResourceBreadcrumbResolver },
+    children: [
+      { path: '', redirectTo: 'overview', pathMatch: 'full' },
+      {
+        path: 'overview',
+        component: PoolResourcePageComponent,
+        data: { breadcrumbs: 'Overview', section: 'overview' }
+      },
+      {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool.module.ts]: You agreed to remove the Advanced Properties section (the `cd-table-key-value` dump). It is still here in the routes and in `pool-resource-page.component.html` `@case ('advanced-properties')`. Please remove the route, the `@case` block, and the sidebar item from `buildSidebarItems` in `pool-resource-sidebar.component.ts`.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts [hunk 23]
```diff
@@ -0,0 +1,319 @@
...
+  private formatBytes(value?: number): string {
+    return this.formatter.format_number(value, 1024, this.binaryUnits, 1);
+  }
+
+  private formatRate(value?: number, binary?: boolean): string {
+    if (binary) {
+      return `${this.formatter.format_number(value, 1024, this.binaryUnits, 1)}/s`;
+    }
+
+    return `${Math.round(value * 10) / 10}/s`;
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-page/pool-resource-page.component.ts:309]: **Bug:** When `value` is `null` or `undefined` (the parameter is `value?: number`), `Math.round(value * 10)` produces `NaN`, so the output is `NaN/s`. Add a null guard:
```ts
if (value == null || !Number.isFinite(value)) {
  return '0/s';
}
return `${Math.round(value * 10) / 10}/s`;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-sidebar/pool-resource-sidebar.component.scss [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+
+.pool-details-layout .sidebar-header {
+  padding-right: var(--cds-spacing-07);
+}
+
+/* Menu is rendered under document.body; scope fixes by menu id. */
+#pool-actions-menu.cds--menu {
+  width: max-content !important;
+  min-width: 12rem;
+  background-color: colors.$white !important;
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-resource-sidebar/pool-resource-sidebar.component.scss]: `colors.$white` is a hardcoded color and will break in dark mode. Use a Carbon theme token instead, e.g. `var(--cds-layer-01)`. Same issue on line 64 with `colors.$gray-10` -- use `var(--cds-background-hover)` instead.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/_pool-resource-overview-card.scss [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+    display: block;
+  }
+
+  .pool-overview-panels {
+    display: grid;
+    grid-template-columns: repeat(2, minmax(0, 1fr));
+    gap: layout.$spacing-06;
+  }
+
+  .pool-detail-panel {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/_pool-resource-overview-card.scss:14]: Commented-out code. Please remove this line. Same issue in `pool-capacity-protection-card.component.scss` line 30.

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Comment by syedali237** [comment]
> Edit Pool looks misaligned from left margin but didnt see any chnges in edit-pool list . Please re check its unrelated to this PR.
> Image

@dnyanee1997 I see this happening with several forms not just this one, when I reload it gets fixed, I guess we should take a look at this issue later..

**Review by afreen23 — APPROVED** [review]
LGTM

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts [hunk 26]
```diff
@@ -46,23 +49,24 @@ const BASE_URL = 'pool';
 })
 export class PoolListComponent extends ListWithDetails implements OnInit {
   @ViewChild(TableComponent)
-  table: TableComponent;
-  @ViewChild('poolUsageTpl', { static: true })
-  poolUsageTpl: TemplateRef<any>;
+  table!: TableComponent;
+  @ViewChild('poolNameTpl', { static: true })
+  poolNameTpl!: TemplateRef<any>;
```
Comment (dnyanee1997) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts:54]: type

**Comment by dnyanee1997** [comment]
Thanks for the work @syedali237 
Pool form seems misaligned from left margin. Seems like not related to this PR. But please have a look onces.

https://github.com/user-attachments/assets/2ba3898d-1e56-4de7-89a3-8c1d422796b7

**Comment by syedali237** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts (hunk 26 above)
Comment (syedali237) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts:54]: Kept it as TemplateRef<any> to stay consistent with existing template refs in this file..
